tests/test_page.py

```diff
@@ -151,10 +152,11 @@ def test_request_types(har_data):
 
     # Check request type lists
     for req in page.get_requests:
-        assert req['request']['method'] == 'GET'
+        assert req.request.method == 'GET'
 
     for req in page.post_requests:
-        assert req['request']['method'] == 'POST'
+        assert req.request.method == 'POST'
```

Comment:
Hmm, you could technically revert these tests back to using the dicts, this would provide extra coverage for your backwards compatible changes.

Reply:
It actually looks really easy with the MutableMapping.
Using:
```
class MimicDict(MutableMapping, ABC):
    """Mixin for functions to mimic a dictionary for backward compatibility"""

    def __getitem__(self, item):
        return self.raw_entry[item]

    def __len__(self):
        return len(self.raw_entry)

    def __delitem__(self, key):
        del self.raw_entry[key]

    def __iter__(self):
        return iter(self.raw_entry)

    def __setitem__(self, key, value):
        self.raw_entry[key] = value
```
Passes all current tests